src/repr/src/adt/numeric.rs

```diff
@@ -762,11 +763,111 @@ impl DecimalLike for Numeric {
     }
 }
 
+/// An encoded packed variant of [`Numeric`].
+///
+/// We uphold the invariant that [`PackedNumeric`] sorts the same as
+/// [`Numeric`].
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct PackedNumeric([u8; Self::SIZE]);
+
+// `as` conversions are okay here because we're doing bit level logic to make
+// sure the sort order of the packed binary is correct. This is implementation
+// is proptest-ed below.
+#[allow(clippy::as_conversions)]
+impl FixedSizeCodec<Numeric> for PackedNumeric {
+    const SIZE: usize = 40;
+
+    fn as_bytes(&self) -> &[u8] {
+        &self.0
+    }
+
+    fn from_bytes(slice: &[u8]) -> Result<Self, String> {
+        let buf: [u8; Self::SIZE] = slice.try_into().map_err(|_| {
+            format!(
+                "size for PackedNumeric is {} bytes, got {}",
+                Self::SIZE,
+                slice.len()
+            )
+        })?;
+        Ok(PackedNumeric(buf))
+    }
+
+    fn from_value(value: Numeric) -> Self {
+        let (digits, exponent, mut bits, lsu) = value.to_raw_parts();
+
+        let mut buf = [0u8; 40];
+
+        // XORing the exponent will ensure negative exponents (smaller) sort before positive
+        // exponents (larger).
+        let exponent = (exponent as u32) ^ 0x8000_0000u32;
+
+        // Naturally the marker bits are in reverse sorted order, except for all positive numbers.
+        // In this case adding 96 and then inverting the bits provides the right sort.
```

Comment:
Not actually adding!

Reply:
Ahh shoot, yeah let me update the comment